### DISCUSSION TITLE
Data-importer: nordigen empty account list

### DISCUSSION THREAD

**Discussion by FrancisLodge65** [firefly-iii/firefly-iii#8358] [Firefly III]
Debug information generated at 2024-01-08 17:06:07 Europe/Rome for Firefly III Data Importer version **1.4.0**.

<table><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr><tr><td>Version</td><td>1.4.0</td></tr><tr><td>System</td><td>PHP 8.3.1, Linux, apache2handler</td></tr></table><table><tr><th colspan="2">App information</th></tr><tr><th>Item</th><th>Value</th></tr><tr><td>Timezone</td><td>Europe/Rome, Europe/Rome</td></tr><tr><td>Environment</td><td>local</td></tr><tr><td>Debug mode</td><td>false, cache 'file'</td></tr><tr><td>Log level</td><td>debug, stack</td></tr><tr><td>Display errors</td><td>Off, ALL errors</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Trusted proxies</td><td></td></tr></table><table><tr><th colspan="2">User information</th></tr><tr><th>Item</th><th>Value</th></tr><tr><td>User agent</td><td>Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36</td></tr></table>

- [x] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)
- [x] I've used [the search](https://github.com/firefly-iii/firefly-iii/issues?q=is%3Aissue) and this has not been requested before.

### Description
After GoCardless login and after accepting the integration with bank returns to importer page but says there is no account in the list and cannot continue. 
This happens only with some banks (Italy - Revolut ok, Italy - Bancoposta/PostePay KO)

![immagine](https://github.com/firefly-iii/firefly-iii/assets/154627058/5532dedd-11e6-4847-a08f-4b8f48af6a43)

If I press continue says
```
Some error(s) occurred:
    You must select at least one account to import from.
```

### Expected behaviour

Continue with import selecting iban.

**Comment by JC5** [comment]
I would like to see the logs of this page to see why it doesn't show you any accounts:

https://docs.firefly-iii.org/how-to/general/debug/

**Reply by FrancisLodge65** [reply]
> [2024-01-09 10:04:58] local.DEBUG: Query parameters are {"client_id":*MY_ID*,"redirect_uri":"http://192.168.1.104:81/callback","response_type":"code","scope":"","state":"5H30K3f98PhFlXEC7B4WIQbficSZcs7NG9ZPz00x","code_challenge":"*LONG_STRING*","code_challenge_method":"S256"}
> [2024-01-09 10:04:58] local.DEBUG: Now redirecting to "http://192.168.1.104/oauth/authorize?" (params omitted)
> [2024-01-09 10:04:59] local.DEBUG: Now at App\Http\Controllers\TokenController::callback
> [2024-01-09 10:04:59] local.DEBUG: State is valid!
> [2024-01-09 10:04:59] local.DEBUG: Params for access token {"form_params":{"grant_type":"authorization_code","client_id":*MY_ID*,"redirect_uri":"http://192.168.1.104:81/callback","code_verifier":"*VERY_LONG_STRING*"}}
> [2024-01-09 10:04:59] local.DEBUG: Will contact "http://192.168.1.104/oauth/token" for a token.
> [2024-01-09 10:04:59] local.DEBUG: Response {"token_type":"Bearer","expires_in":1209600,"access_token":"*VERY_LONG_STRING*"}
> [2024-01-09 10:04:59] local.DEBUG: Return redirect  to "http://192.168.1.104:81"
> [2024-01-09 10:04:59] local.DEBUG: Now in App\Http\Controllers\IndexController::index
> [2024-01-09 10:04:59] local.DEBUG: App\Services\Shared\Authentication\SecretManager::hasValidSecrets
> [2024-01-09 10:05:00] local.DEBUG: Now at App\Http\Controllers\TokenController::doValidate
> [2024-01-09 10:05:00] local.DEBUG: Now trying to authenticate with Firefly III at http://192.168.1.104
> [2024-01-09 10:05:00] local.DEBUG: isReadyForStep returns true because $flow is null
> [2024-01-09 10:05:00] local.DEBUG: Now at App\Services\Spectre\AuthenticationValidator::validate
> [2024-01-09 10:05:00] local.DEBUG: No Spectre App ID in hasAppId() session, will return config variable.
> [2024-01-09 10:05:00] local.DEBUG: No Spectre secret in hasSecret(), will return config variable.
> [2024-01-09 10:05:00] local.DEBUG: isReadyForStep returns true because $flow is null
> [2024-01-09 10:05:00] local.DEBUG: Now at App\Services\Nordigen\AuthenticationValidator::validate
> [2024-01-09 10:05:00] local.DEBUG: No Nordigen ID in hasId() session, will return config variable.
> [2024-01-09 10:05:00] local.DEBUG: No Nordigen key in hasKey() session, will return config variable.
> [2024-01-09 10:05:00] local.DEBUG: Result is {"result":"OK","message":null}
> [2024-01-09 10:05:01] local.DEBUG: Now in App\Http\Controllers\IndexController::postIndex
> [2024-01-09 10:05:01] local.DEBUG: nordigen is a valid flow, redirect to authenticate.
> [2024-01-09 10:05:01] local.DEBUG: Now in AuthenticateController, calling middleware.
> [2024-01-09 10:05:01] local.DEBUG: Now at App\Services\Nordigen\AuthenticationValidator::validate
> [2024-01-09 10:05:01] local.DEBUG: No Nordigen ID in hasId() session, will return config variable.
> [2024-01-09 10:05:01] local.DEBUG: No Nordigen key in hasKey() session, will return config variable.
> [2024-01-09 10:05:01] local.DEBUG: No Nordigen key in hasKey() session, will return config variable.
> [2024-01-09 10:05:01] local.DEBUG: No Nordigen ID in hasId() session, will return config variable.
> [2024-01-09 10:05:05] local.DEBUG: Now in AuthenticateController, calling middleware.
> [2024-01-09 10:05:05] local.DEBUG: Now in AuthenticateController, calling middleware.
> [2024-01-09 10:05:05] local.DEBUG: Now at App\Services\Nordigen\AuthenticationValidator::validate
> [2024-01-09 10:05:05] local.DEBUG: Now at App\Services\Nordigen\TokenManager::hasValidRefreshToken
> [2024-01-09 10:05:05] local.DEBUG: No Nordigen refresh token, so return false.
> [2024-01-09 10:05:05] local.DEBUG: No refresh token, so return false.
> [2024-01-09 10:05:05] local.DEBUG: Now at App\Services\Nordigen\TokenManager::getNewTokenSet
> [2024-01-09 10:05:06] local.DEBUG: Now at App\Http\Controllers\Import\UploadController::index
> [2024-01-09 10:05:06] local.DEBUG: Going to check directory for config files: /var/www/html/data-importer/storage/configurations
> [2024-01-09 10:05:06] local.DEBUG: List of files:
> [2024-01-09 10:05:08] local.DEBUG: Now at App\Http\Controllers\Import\UploadController::upload
> [2024-01-09 10:05:08] local.DEBUG: Now at App\Http\Controllers\Import\Nordigen\SelectionController::index
> [2024-01-09 10:05:08] local.DEBUG: Request parameters will be set to:
> [2024-01-09 10:05:08] local.DEBUG: authenticatedGet(https://ob.nordigen.com/api/v2/institutions/)
> [2024-01-09 10:05:20] local.DEBUG: Now at App\Http\Controllers\Import\Nordigen\SelectionController::postIndex
> [2024-01-09 10:05:20] local.DEBUG: storeContent: Stored 907 bytes in file "dd6243fed4fc995495f73a82b54d6ae79d1fc6dfb50c68e33137788ded58fe08"
> [2024-01-09 10:05:20] local.DEBUG: Now at App\Http\Controllers\Import\Nordigen\LinkController::build
> [2024-01-09 10:05:20] local.DEBUG: Request parameters will be set to:
> [2024-01-09 10:05:20] local.DEBUG: Now at App\Services\Nordigen\Request\PostNewUserAgreement::post
> [2024-01-09 10:05:20] local.DEBUG: Now at App\Services\Nordigen\Request\Request::authenticatedJsonPost
> [2024-01-09 10:05:20] local.DEBUG: Returned from POST:  {"id":"cf9bb776-aeff-4845-a267-6f3258e18f0b","created":"2024-01-09T09:05:20.807610Z","institution_id":"POSTE_ITALIANE_BANCO_POSTA_BPPIITRRXXX","max_historical_days":90,"access_valid_for_days":90,"access_scope":["balances","details","transactions"],"accepted":null}
> [2024-01-09 10:05:20] local.DEBUG: Request parameters will be set to:
> [2024-01-09 10:05:20] local.DEBUG: Reference is "1b06988a-ae43-4618-8446-5bd2594c2ec2"
> [2024-01-09 10:05:20] local.DEBUG: Now at App\Services\Nordigen\Request\PostNewRequisitionRequest::post
> [2024-01-09 10:05:20] local.DEBUG: Now at App\Services\Nordigen\Request\Request::authenticatedJsonPost
> [2024-01-09 10:05:21] local.DEBUG: Returned from POST:  {"id":"f53fcd20-ef98-4dbf-bfbe-e5e9ad410794","created":"2024-01-09T09:05:21.133819Z","redirect":"http://192.168.1.104:81/import/link-nordigen/callback","status":"CR","institution_id":"POSTE_ITALIANE_BANCO_POSTA_BPPIITRRXXX","agreement":"cf9bb776-aeff-4845-a267-6f3258e18f0b","reference":"1b06988a-ae43-4618-8446-5bd2594c2ec2","accounts":[],"link":"https://ob.nordigen.com/psd2/start/f53fcd20-ef98-4dbf-bfbe-e5e9ad410794/POSTE_ITALIANE_BANCO_POSTA_BPPIITRRXXX","ssn":null,"account_selection":false,"redirect_immediate":false}
> [2024-01-09 10:05:21] local.DEBUG: Got a new requisition with id "f53fcd20-ef98-4dbf-bfbe-e5e9ad410794"
> [2024-01-09 10:05:21] local.DEBUG: Status: CR, returned reference: "1b06988a-ae43-4618-8446-5bd2594c2ec2"
> [2024-01-09 10:05:21] local.DEBUG: Will now redirect the user to https://ob.nordigen.com/psd2/start/f53fcd20-ef98-4dbf-bfbe-e5e9ad410794/POSTE_ITALIANE_BANCO_POSTA_BPPIITRRXXX
> [2024-01-09 10:05:51] local.DEBUG: Now at App\Http\Controllers\Import\Nordigen\LinkController::callback
> [2024-01-09 10:05:51] local.DEBUG: Reference is "1b06988a-ae43-4618-8446-5bd2594c2ec2"
> [2024-01-09 10:05:51] local.DEBUG: Now at App\Http\Controllers\Import\ConfigurationController::index
> [2024-01-09 10:05:51] local.DEBUG: Now in App\Support\Internal\CollectsAccounts::getNordigenAccounts
> [2024-01-09 10:05:51] local.DEBUG: Request parameters will be set to:
> [2024-01-09 10:05:51] local.DEBUG: authenticatedGet(https://ob.nordigen.com/api/v2/requisitions/f53fcd20-ef98-4dbf-bfbe-e5e9ad410794/)
> [2024-01-09 10:05:51] local.DEBUG: ListAccountsResponse: {"id":"f53fcd20-ef98-4dbf-bfbe-e5e9ad410794","created":"2024-01-09T09:05:21.133819Z","redirect":"http://192.168.1.104:81/import/link-nordigen/callback","status":"RJ","institution_id":"POSTE_ITALIANE_BANCO_POSTA_BPPIITRRXXX","agreement":"cf9bb776-aeff-4845-a267-6f3258e18f0b","reference":"1b06988a-ae43-4618-8446-5bd2594c2ec2","accounts":[],"link":"https://ob.nordigen.com/psd2/start/f53fcd20-ef98-4dbf-bfbe-e5e9ad410794/POSTE_ITALIANE_BANCO_POSTA_BPPIITRRXXX","ssn":null,"account_selection":false,"redirect_immediate":false}
> [2024-01-09 10:05:51] local.DEBUG: Found 0 Nordigen accounts.
> [2024-01-09 10:05:51] local.DEBUG: Now merging Nordigen account lists.
> [2024-01-09 10:05:51] local.DEBUG: Now in App\Services\Shared\Model\ImportServiceAccount::convertNordigenArray
> [2024-01-09 10:05:52] local.DEBUG: Method App\Http\Controllers\Import\ConfigurationController::phpDate
> 

Here it is DataImporter's logs! Thanks!

**Reply by JC5** [reply]
I see this line:

>  [2024-01-09 10:05:51] local.DEBUG: Found 0 Nordigen accounts.

Perhaps Nordigen / GoCardless has no access to your account? I'm not really sure.

**Reply by FrancisLodge65** [reply]
I do think this is the problem. Trying with Revolut bank it works without problem...

**Reply by JC5** [reply]
Mmmm :(

Perhaps GoCardless / Nordigen support can help here?

**Comment by Elekam** [comment]
I'm facing the same problem with the same "Found 0 Nordigen Accounts" line.
